### PR TITLE
Feature/APIs#11

### PR DESCRIPTION
**Pull request by ashishkava**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  // set locals, only providing error in development
+  res.locals.message = err.message;
+  res.locals.error = req.app.get('env') === 'development' ? err : {};
+
+  // render the error page
+  res.status(err.status || 500);
+  res.render('error');
+});
+
+module.exports = app;
```
Comment (jaimin45) [inline app.js:52]: Add line

File: index.jade [hunk 2]
```diff
@@ -0,0 +1,11 @@
+extends layout
+
+block content
+  h1= title
+  p Welcome to Express Generator 
+
+  h2= line 
+  p Use the application generator tool, express-generator , to quickly create an application skeleton.
+  
+  h3= Title
```
Comment (jaimin45) [inline index.jade:10]: add line

**Review by jaimin45 — CHANGES_REQUESTED** [review]

**Review by JainishaParmar — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (jaimin45) [inline app.js:52]: add line

File: index.jade [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  p Quick Start
```
Comment (jaimin45) [inline index.jade:11]: add line

File: name.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerYamlDocument));
+
+const port = 8080;
+app.listen(port, () => {
+  // eslint-disable-next-line no-console
+  console.log(`A node js API is listening on port: ${port}`);
+});
+
+module.export = app;
```
Comment (jaimin45) [inline name.js:16]: add line

File: tutorial.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+} = require("../controllers/tutorialsControllers");
+
+/* GET users listing. */
+router.get("/", tutorialsAll);
+router.get("/:tutorialId", tutorialSearchById);
+router.post("/", tutorialsCreate);
+router.put("/:tutorialId", updateTutorials);
+router.delete("/:tutorialId", deleteTutorials);
+
+module.export = router;
```
Comment (jaimin45) [inline tutorial.js:17]: add line

File: tutorial.schema.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      required: true,
+    },
+    updatedAt: {
+      type: Date,
+      default: Date.now,
+      required: true,
+    },
+  });
+  
+  module.exports = mongoose.model("Tutorial1", tutorialSchema.js);
```
Comment (jaimin45) [inline tutorial.schema.js:24]: add line

File: app.js (hunk 1 above)
Comment (JainishaParmar) [inline app.js:52]: enter new line

**Review by jaimin45 — CHANGES_REQUESTED** [review]

File: tutorial.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  updateTutorials,
+  deleteTutorials,
+} = require("../controllers/tutorialsControllers");
+
+/* GET users listing. */
+router.get("/", tutorialsAll);
+router.get("/:tutorialId", tutorialSearchById);
+router.post("/", tutorialsCreate);
+router.put("/:tutorialId", updateTutorials);
+router.delete("/:tutorialId", deleteTutorials);
```
Comment (JainishaParmar) [inline tutorial.js:15]: tutorial id should `id`

File: users.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+const router = express.Router();
+
+/* GET users listing. */
+router.get('/', function(req, res, next) {
+  dsfsdf
```
Comment (JainishaParmar) [inline users.js:6]: remove this `dsfsdf`

File: www.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+const http = require('http');
+const debug = require('debug')('nodeJsDemo:server');
+const app = require('../app');
+
+function normalizePort(val) {
+  const port = parseInt(val, 10);
+
+  // if (isNaN(port)) {
+  //   return val;
+  // }
```
Comment (JainishaParmar) [inline www.js:10]: remove this

**Review by JainishaParmar — CHANGES_REQUESTED** [review]

**Review by jaimin45 — APPROVED** [review]

File: swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+            $ref: "#/definitions/Tutorial"
+      produces:
+        - application/json
+      responses:
+        "200":
+          description: New user is created
+          schema:
+            $ref: "#/definitions/Tutorial"
+        "404":
+          description: "Invalid input"
```
Comment (JainishaParmar) [inline swagger.yaml:56]: add 500 response in all APIs and add schema ```$ref: "#/definitions/500Response"```

File: swagger.yaml [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+      tags:
+        - Tutorials
+      summary: Get all tutorials in system
+      responses:
+        "200":
+          description: OK
+          schema:
+            $ref: "#/definitions/Tutorial"
+        "404":
+          description: "Not found"
```
Comment (JainishaParmar) [inline swagger.yaml:36]: add schema ```$ref: "#/definitions/404Response"```

File: tutorial.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+  updateTutorials,
+  deleteTutorials,
+} = require("../controllers/tutorialsControllers");
+
+/* GET users listing. */
+router.get("/", tutorialsAll);
+router.get("/:tutorialid", tutorialSearchByid);
+router.post("/", tutorialsCreate);
+router.put("/:tutorialid", updateTutorials);
+router.delete("/:tutorialid", deleteTutorials);
```
Comment (JainishaParmar) [inline tutorial.js:15]: change ```tutorialid``` to ```id```

File: tutorial.schema.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    createdAt: {
+      type: Date,
+      default: Date.now,
+      required: true,
+    },
+    updatedAt: {
+      type: Date,
+      default: Date.now,
+      required: true,
+    },
```
Comment (JainishaParmar) [inline tutorial.schema.js:21]: please remove id createdAt and updatedAt it should autogenerated and add title and description

File: Eslintrc.json [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    },
+    "extends": ["plugin:prettier/recommended", "airbnb-base"],
+    "parserOptions": {
+        "ecmaVersion": 8080,
+        "sourceType": "module"
+    },
+    "plugins": ["prettier", "spellcheck"],
+    "rules": {
+      "quotes": "off",
+      "prettier/prettier": "error",
```
Comment (jaimin45) [inline Eslintrc.json:14]: remove prettier

**Review by jaimin45 — CHANGES_REQUESTED** [review]

**Review by JainishaParmar — CHANGES_REQUESTED** [review]

File: app.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+const path = require('path');
+const cookieParser = require('cookie-parser');
+const swaggerUI = require('swagger-ui-express');
+const swaggerYaml = require('yamljs');
+const logger = require('morgan');
+const { promise } = require('./init/db');
+
+const indexRouter = require('./routes/index');
+const usersRouter = require('./routes/users');
+const TutorialRouter = require('./routes/tutorial.routes');
```
Comment (kapilbarad) [inline app.js:12]: Bane should be `tutorialRouter`

File: name.js [hunk 16]
```diff
@@ -0,0 +1,16 @@
+const express = require("express");
```
Comment (kapilbarad) [inline name.js:1]: Please remove this file as It's not being used anywhere.

File: swagger.yaml [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+    description: API for user in the system
+schemes:
+  - http
+consumes:
+  - application/json
+produces:
+  - application/json
+paths:
+  /tutorials:
+    parameters:
```
Comment (kapilbarad) [inline swagger.yaml:19]: No need of this parameters here as we don't have id in path parameters for Create and Get routes.

**Review by kapilbarad — CHANGES_REQUESTED** [review]

**Review by jaimin45 — APPROVED** [review]

**Review by JainishaParmar — APPROVED** [review]

**Review by JainishaParmar — APPROVED** [review]

**Review by kapilbarad — APPROVED** [review]

**Review by JainishaParmar — APPROVED** [review]

**Pull request merged**